### PR TITLE
Add flags to backend

### LINKED ISSUE #6285: ✨ Feature flags for the backend
## ✨ Feature

We have a need to start adjusting the behaviour of the backend based on feature flags.

## 🕵️ Details

Our goals for this:
- Use the exact same flags as the frontend
- Use the .env file as we are currently used to already
- Start with the simplest possible implementation - no new plugins or packages

## ✅ Acceptance Criteria

- [ ] A simple proof-of-concept implementation using one of the current three feature flags showing a pattern that can be followed

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #6285 

## 👋 Introduction

This branch adds feature flags to the PHP backend.

## 🧪 Testing

1. Recopy the .env.example file to .env (or rerun setup.sh)
2. Authenticate or update the AUTH_DEFAULT_USER value
3. Run a simple isComplete query in Graphiql
```graphql
 query MeQuery {
  me {
    isProfileComplete
  }
}
```
4. Check the `api/storage/logs/laravel.log` file for the logging message

- [ ] Flags set to true log as true
- [ ] Flags set as false log as false
- [ ] Missing flags log as false

## 📸 Screenshot

image

### CODE REVIEW

File: api/.env.example [hunk 1]
```diff
@@ -84,11 +84,15 @@ FRESHDESK_API_TICKET_TAG=
...
 GCNOTIFY_TEMPLATE_TEST_EMAIL="c386053b-e65c-477e-8f09-da19c2e9dea0"
 GCNOTIFY_TEMPLATE_TEST_SMS="3525f0e3-845a-4001-b456-7cb02c13c69f"
 GCNOTIFY_TEMPLATE_TEST_BULK_EMAIL="90fa07f5-fbe8-402e-9547-f3259159a2da"
 GCNOTIFY_TEMPLATE_TEST_BULK_SMS="d662aac4-08bb-4917-860f-cc9999f4648d"
 
+# Feature flags
+FEATURE_ONGOING_RECRUITMENTS=true
+FEATURE_APPLICANT_DASHBOARD=true
+FEATURE_APPLICATION_REVAMP=true
+FEATURE_PSAC_STRIKE=true
```
Comment (esizer) [inline api/.env.example:98]: Just thinking here, nothing to action. But, I really wish we had  central location for these in development 😓

File: api/config/feature.php
Comment (esizer) [inline api/config/feature.php:1]: Nice, I really like storing these in config for easy access! 👍

File: api/config/feature.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    'applicant_dashboard' => (bool) env('FEATURE_APPLICANT_DASHBOARD', false),
+
+    /*
+    |--------------------------------------------------------------------------
+    | Application Revamp
+    |--------------------------------------------------------------------------
+    |
+    | AKA Application V6, AKA New Application Flow.  A wizard-style application
+    | process.
```
Comment (esizer) [inline api/config/feature.php:33]: 🤣

**Review by esizer — APPROVED** [review]
Wow, very clean. Not much to say here but great work 👍

File: api/.env.example (hunk 1 above)
Comment (petertgiles) [reply api/.env.example:98]: Funny you should mention that...
https://github.com/GCTC-NTGC/gc-digital-talent/discussions/6375

File: api/config/feature.php (hunk 2 above)
Comment (petertgiles) [reply api/config/feature.php:33]: :wink:

**Comment by petertgiles** [comment]
Confirmed it worked in Azure, too.
image

**Pull request merged**